.github/workflows/merge-flow.yaml

```diff
@@ -0,0 +1,142 @@
+name: Common merge flow
+on:
+  workflow_call:
+    inputs:
+      go-version:
+        description: go version
+        required: true
+        type: string
+      upstream:
+        description: Upstream repo path in owner/repo format
+        required: true
+        type: string
+      downstream:
+        description: Downstream repo path in owner/repo format
+        required: true
+        type: string
+      downstream-branch:
+        description: Downstream branch to create PR
+        required: false
+        default: master
+        type: string
+      sandbox:
+        description: Sandbox repo path in owner/repo format. Used a base to create PR against downstream.
+        required: true
+        type: string
+      theirs:
+        description: List of files to be reset using upstream content on merge conflict.
+        required: false
+        default: ''
+        type: string
+      ours:
+        description: List of files to be reset using downstream content on merge conflict.
+        required: false
+        default: ''
+        type: string
+    secrets:
+      cloner-app-id:
+        description: Github ID of cloner app
+        required: true
+      cloner-app-private-key:
+        description: Github private key of cloner app
+        required: true
+      pr-app-id:
+        description: Github ID of PR creation app
+        required: true
+      pr-app-private-key:
+        description: Github private key of PR creation app
+        required: true
+
+jobs:
+  merge:
+    runs-on: ubuntu-latest
+    name: perform merge operation
+    steps:
+      - name: get latest upstream tag
+        id: upstream
+        uses: pozetroninc/github-action-get-latest-release@master
+        with:
+          repository: ${{ inputs.upstream }}
+          excludes: prerelease, draft
+      - uses: actions/checkout@v2
+        with:
+          repository: ${{ inputs.downstream }}
+          fetch-depth: 0
+          ref: ${{ inputs.downstream-branch }}
+      - name: fetch upstream
+        run: |
+          echo latest tag is ${{ steps.upstream.outputs.release }}
+      - name: fetch upstream
+        run: |
+          git config user.name 'github-actions[bot]'
+          git config user.email 'github-actions[bot]@users.noreply.github.com'
+          git config --global core.editor "/bin/true"
+          git fetch https://github.com/${{ inputs.upstream }} --tags
+      - name: merge upstream
+        id: merge
+        run: |
+          git merge refs/tags/${{ steps.upstream.outputs.release }} --no-edit || echo '::set-output name=MERGE_CONFLICT::true'
+      - name: resolve conflict using theirs
+        if: ${{ steps.merge.outputs.MERGE_CONFLICT == 'true' && inputs.theirs != ''}}
```

Comment:
I feel it may be more readable if we used `input.merge_strategy = "theirs|ours"` and then specified a list of files so that the code will read something like ... 

```c
if merge_conflict {
   git checkout --${{inputs.merge_strategy}} inputs.files
}
```

Reply:
@sthaha The flow uses default merge strategy, however during a merge conflict it provides options to reset the contents of files certain files using `theirs` and `ours` inputs. `their` resets the given file content using `upstream` and `ours` does the opposite. So `theirs` and `ours` are not mutually exclusive, can exist at the same time.
for e.g, consider KSM merge, when conflict occurs, we know few files has to be reset using upstream content(CHANGELOG) and few using downstream(OWNERS).